hwcomposer/qeglfsbackingstore.cpp

```diff
@@ -145,7 +147,7 @@ void QEglFSBackingStore::flush(QWindow *window, const QRegion &region, const QPo
         QRect imageRect = m_image.rect();
 
         QRegion fixed;
-        foreach (const QRect &rect, m_dirty.rects()) {
+        for (const QRect &rect : m_dirty) {
```

Comment:
just curious - what kind of compile error or warning `foreach` is producing under Qt 5.15?

Reply:
It's in the commit message of 14caadf8c70e405d36f2dc56e8c3cdc8c8d3463f as well as my initial PR message above, just search for `foreach` and you should see it :) Either way the old form produces the following warning:
```
'QVector<QRect> QRegion::rects() const' is deprecated: Use begin()/end() instead [-Wdeprecated-declarations]
```

Reply:
ok, thanks

Reply:
Might be offtopic but foreach should not be used anymore, it prefered to use C++11for "each".